File: 백준/Silver/1406. 에디터/에디터.java
//https://www.acmicpc.net/problem/1406
//소문자 lowerCase, 대문자 upperCase
import java.util.*;
import java.io.*;

public class Main {
    private static Stack<Character> leftStack = new Stack<>();
    private static Stack<Character> rightStack = new Stack<>();

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String str = br.readLine();

        if(isAllLowerCase(str)){
            for (char c : str.toCharArray()) {
                leftStack.push(c);
            }

            int M = Integer.parseInt(br.readLine());
            while(M-- > 0){
                String[] command = br.readLine().split(" ");
                switch (command[0]) {
                    case "L":
                        moveCursorLeft();
                        break;
                    case "P":
                        addCharacter(command[1].charAt(0));
                        break;
                    case "B":
                        removeCursorLeft();
                        break;
                    case "D":
                        moveCursorRight();
                        break;
                }
            }
            bw.write(getText());
            bw.flush();
        }
    }

    public static boolean isAllLowerCase(String str) {
        for (int i = 0; i < str.length(); i++) {
            if (!Character.isLowerCase(str.charAt(i))) {
                return false;
            }
        }
        return true;
    }

    public static void addCharacter(char c) {
        leftStack.push(c);
    }
// 왼쪽 스택의 맨 위에 있는 상자를 오른쪽 스택의 맨 위로 옮겨놓는다.
    public static void moveCursorLeft() {
        if (!leftStack.isEmpty()) {
            rightStack.push(leftStack.pop());
        }
    }
    public static void removeCursorLeft() {
        if (!leftStack.isEmpty()) {
            leftStack.pop();
        }
    }
    // 오른쪽 스택의 맨 위에 있는 상자를 왼쪽 스택의 맨 위로 옮겨놓는다.
    public static void moveCursorRight() {
        if (!rightStack.isEmpty()) {
            leftStack.push(rightStack.pop());
        }
    }
    public static String getText() {
        StringBuilder text = new StringBuilder();
        for (Character c : leftStack) {
            text.append(c);
        }
        while (!rightStack.isEmpty()) {
            text.append(rightStack.pop());
        }
        return text.toString();
    }
}

File: 백준/Silver/1406. 에디터/에디터.py
import sys

# 문자열 스택
left_stack = list(sys.stdin.readline().strip())
right_stack = []

# 명령어 입력
num_commands = int(sys.stdin.readline().strip())

for _ in range(num_commands):
    command = sys.stdin.readline().split()

    if command[0] == 'L' and left_stack:
        right_stack.append(left_stack.pop())
    elif command[0] == 'D' and right_stack:
        left_stack.append(right_stack.pop())
    elif command[0] == 'B' and left_stack:
        left_stack.pop()
    elif command[0] == 'P':
        left_stack.append(command[1])

# right_stack을 뒤집어서 left_stack과 합친 후 문자열로 변환하여 출력
right_stack.reverse()
print(''.join(left_stack + right_stack))
